src/main/java/org/skriptlang/skript/bukkit/spark/SparkUtils.java

```diff
@@ -0,0 +1,149 @@
+	 *
+	 * @param window the window for which to gather statistics from.
+	 * @return the MSPT info block.
+	 */
+	public static @Nullable DoubleAverageInfo mspt(MillisPerTick window) {
+		GenericStatistic<DoubleAverageInfo, MillisPerTick> mspt = mspt();
+		return mspt.poll(window);
+	}
+
+	/**
+	 * Get the MSPT stat info block for the given windows.
+	 * {@link #supportsMspt()} should be checked prior to using this.
+	 *
+	 * @param windows the windows for which to gather statistics from.
+	 * @return the MSPT info blocks.
+	 */
+	public static DoubleAverageInfo @Nullable [] mspt(MillisPerTick... windows) {
+		GenericStatistic<DoubleAverageInfo, MillisPerTick> mspt = mspt();
+		return Arrays.stream(windows).map(mspt::poll).toArray(DoubleAverageInfo[]::new);
+	}
+
+	private static DoubleStatistic<CpuUsage> cpuProcess() {
+		if (cpuProcess == null)
+			cpuProcess = spark().cpuProcess();
+		return cpuProcess;
+	}
+
+	/**
+	 * Get the process CPU usage for the given window.
+	 * @param window the window for which to gather statistics from.
+	 * @return the process cpu usage.
+	 */
+	public static double cpuProcess(CpuUsage window) {
+		return cpuProcess().poll(window);
+	}
+
+	/**
+	 * Get the process CPU usage for the given windows.
+	 * @param windows the windows for which to gather statistics from.
+	 * @return the process cpu usages.
+	 */
+	public static Double[] cpuProcess(CpuUsage... windows) {
+		DoubleStatistic<CpuUsage> cpu = cpuProcess();
+		return Arrays.stream(windows).map(cpu::poll).toArray(Double[]::new);
+	}
+
+	private static DoubleStatistic<CpuUsage> cpuSystem() {
+		if (cpuSystem == null)
+			cpuSystem = spark().cpuSystem();
+		return cpuSystem;
+	}
+
+	/**
+	 * Get the system CPU usage for the given window.
+	 * @param window the window for which to gather statistics from.
+	 * @return the system cpu usage.
+	 */
+	public static double cpuSystem(CpuUsage window) {
+		return cpuSystem().poll(window);
+	}
+
+	/**
+	 * Get the system CPU usage for the given windows.
+	 * @param windows the windows for which to gather statistics from.
+	 * @return the system cpu usages.
+	 */
+	public static Double[] cpuSystem(CpuUsage... windows) {
+		DoubleStatistic<CpuUsage> cpu = cpuSystem();
+		return Arrays.stream(windows).map(cpu::poll).toArray(Double[]::new);
+	}
+
+	private static DoubleStatistic<TicksPerSecond> tps() {
+		if (tps == null)
+			tps = spark().tps();
+		return tps;
+	}
+
+	/**
+	 * Get the TPS for the given windows.
+	 * @param window the window for which to gather statistics from.
+	 * @return the TPS value.
+	 */
+	public static double tps(TicksPerSecond window) {
+		return tps().poll(window);
+	}
+
+	/**
+	 * Get the TPS for the given windows.
+	 * @param windows the windows for which to gather statistics from.
+	 * @return the TPS values.
+	 */
+	public static Double[] tps(TicksPerSecond... windows) {
+		DoubleStatistic<TicksPerSecond> tps = tps();
+		return Arrays.stream(windows).map(tps::poll).toArray(Double[]::new);
```

Comment:
helper method? this is repeated ~~3~~ 4 times, just a suggestion though.

Edit: this is repeated in ExprMSPT too.

Reply:
I'm not sure how beneficial a helper would be since it's not going to save much space/complexity but it's worth trying